### PR TITLE
[FLINK-2559] Fix Javadoc Code Examples

### PR DESCRIPTION
**Pull request by hczerpak**
Initially I've made only handful of fixes to javadocs, replacing few @ with  {@literal @}. Running mvn javadoc:javadoc revealed lots of javadoc  problems:

- broken html tags: `<br/>, <p/>, </br>.`
- unclosed html tags e.g. `<i>...<i>`
- lots of > and < characters used directly in javadoc
- source code examples not wrapped with {@code }
- incorrect references to classes, methods in @see or @link tags
- @throws tags when no exception is being thrown (or different)
- no @throws when exception is being thrown from method
- typos 

Unfortunately Travis doesn't run javadocs compilation and it will not show that it actually works

### CODE REVIEW

File: flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/util/SplitStreamType.java [hunk 1]
```diff
@@ -20,10 +20,11 @@
 import org.apache.flink.streaming.api.datastream.DataStream;
 
 /**
- * Used by {@link org.apache.flink.storm.wrappers.AbstractStormCollector AbstractStormCollector} to wrap
+ * Used by org.apache.flink.storm.wrappers.AbstractStormCollector to wrap
```
Comment (fhueske) [inline flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/util/SplitStreamType.java]: Why did you remove this link?

File: flink-contrib/flink-tweet-inputformat/src/main/java/org/apache/flink/contrib/tweetinputformat/model/tweet/entities/Entities.java [hunk 2]
```diff
@@ -21,8 +21,8 @@
 import java.util.List;
 
 /**
- * Entities which have been parsed out of the text of the
- * {@link package org.apache.flink.contrib.tweetinputformat.model.tweet.Tweet}.
+ * entities which have been parsed out of the text of the
```
Comment (fhueske) [inline flink-contrib/flink-tweet-inputformat/src/main/java/org/apache/flink/contrib/tweetinputformat/model/tweet/entities/Entities.java]: Why lowercase?

File: flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java [hunk 3]
```diff
@@ -314,7 +314,7 @@ public void remove() {
 
 	/**
 	 * Retrieves the vertex in-degree (number of in-coming edges).
-	 * @return The in-degree of this vertex if the {@link IterationConfiguration#setOptDegrees(boolean)}
+	 * @return The in-degree of this vertex
```
Comment (fhueske) [inline flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java]: the `if` part is missing

File: flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java [hunk 4]
```diff
@@ -327,7 +327,7 @@ void setInDegree(long inDegree) {
 
 	/**
 	 * Retrieve the vertex out-degree (number of out-going edges).
-	 * @return The out-degree of this vertex if the {@link IterationConfiguration#setOptDegrees(boolean)}
+	 * @return The out-degree of this vertex
```
Comment (fhueske) [inline flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java]: the `if` part is missing

File: flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexUpdateFunction.java [hunk 5]
```diff
@@ -206,7 +206,7 @@ void setOutput(Vertex<K, VV> outVal, Collector<Vertex<K, VV>> out) {
 
 	/**
 	 * Retrieves the vertex in-degree (number of in-coming edges).
-	 * @return The in-degree of this vertex if the {@link IterationConfiguration#setOptDegrees(boolean)}
+	 * @return The in-degree of this vertex
```
Comment (fhueske) [inline flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexUpdateFunction.java]: the `if` part is missing

File: flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexUpdateFunction.java [hunk 6]
```diff
@@ -219,7 +219,7 @@ void setInDegree(long inDegree) {
 
 	/**
 	 * Retrieve the vertex out-degree (number of out-going edges).
-	 * @return The out-degree of this vertex if the {@link IterationConfiguration#setOptDegrees(boolean)}
+	 * @return The out-degree of this vertex
```
Comment (fhueske) [inline flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexUpdateFunction.java]: the `if` part is missing

**Comment by fhueske** [comment]
Hi @hczerpak, thanks a lot for the thorough cleaning of the JavaDocs!
I spotted just a few things that should be improved.
The commit history of your PR is also a bit messy. Would you mind to rebase it to get rid of the merge commits?

Thanks again, Fabian

File: flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/util/SplitStreamType.java (hunk 1 above)
Comment (hczerpak) [inline flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/util/SplitStreamType.java]: Javadoc couldn't link that class. I'm suspecting because AbstractStormCollector has package access and this link has been created from outside of that package. The least I could do was to leave raw path to that class.

File: flink-contrib/flink-tweet-inputformat/src/main/java/org/apache/flink/contrib/tweetinputformat/model/tweet/entities/Entities.java (hunk 2 above)
Comment (hczerpak) [inline flink-contrib/flink-tweet-inputformat/src/main/java/org/apache/flink/contrib/tweetinputformat/model/tweet/entities/Entities.java]: Good point. Didn't intend to lower that case.

File: flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java (hunk 3 above)
Comment (hczerpak) [inline flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java]: There is no IterationConfiguration#setOptDegrees(boolean) any more and if you look at the source code which this javadoc relates to it. It looked like copy-paste comment from somewhere else.

Comment (fhueske) [inline flink-libraries/flink-gelly/src/main/java/org/apache/flink/graph/spargel/MessagingFunction.java]: I see, but the next line should be removed then as well, right?

**Comment by hczerpak** [comment]
Sure. I appreciate your comments. It was not always straightforward what to do with errors. Could you please take a look at my comments above and say what you think?

thanks

**Comment by hczerpak** [comment]
Sorry for missclick. Didn't intend to close and reopen

**Comment by fhueske** [comment]
No worries :-)
Thanks for the quick response!

File: flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/util/SplitStreamType.java (hunk 1 above)
Comment (fhueske) [inline flink-contrib/flink-storm/src/main/java/org/apache/flink/storm/util/SplitStreamType.java]: Sounds good to me.

**Comment by fhueske** [comment]
Thanks for the update.
Looks good to merge.

**Comment by hczerpak** [comment]
I'm sorry @fhueske but honestly I don't know how to squash the history. After few tires I've given up yesterday, it didn't happen correctly.

**Comment by fhueske** [comment]
OK, I can do it as well.

**Comment by fhueske** [comment]
I will merge this PR.
